test/integration/helpers/wait-for-store-resolvers.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { act } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { withFakeTimers } from './with-fake-timers';
+
+/**
+ * Executes a function that triggers store resolvers and waits for them to be finished.
+ *
+ * Asynchronous store resolvers leverage `setTimeout` to run at the end of
+ * the current JavaScript block execution. In order to prevent "act" warnings
+ * triggered by updates to the React tree, we manually tick fake timers and
+ * await the resolution of the current block execution before proceeding.
+ *
+ * @param {Function} fn Function that to trigger.
+ *
+ * @return {*} The result of the function call.
+ */
+export async function waitForStoreResolvers( fn ) {
+	return withFakeTimers( async () => {
+		const result = fn();
+
+		// Advance all timers allowing store resolvers to resolve.
+		act( () => jest.runAllTimers() );
```

Comment:
An idea: Maybe in integration testing, we can [opt-out](https://github.com/reactwg/react-18/discussions/102) of the act warning?

Reply:
I did try setting this var but it didn't get rid of the warnings - feel free to give it a try though, maybe I did something wrong.